community/values/src/main/java/org/neo4j/values/storable/UTF8StringValue.java

```diff
@@ -159,18 +159,18 @@ public int computeHash()
     @Override
     public TextValue substring( int start, int end )
     {
-        assert start > 0;
-        assert end > start && end < length();
+        int s = Math.min( start, length() );
+        int e = Math.min( end, length() );
```

Comment:
Hmm... this still computes the length. And *twice* non the less. So no improvements there...